frontend/src/helpers/__tests__/serverError.spec.js

```diff
@@ -0,0 +1,160 @@
+    const i18n = { te, tc }
+
+    beforeEach(() => {
+      delete i18n.locale
+      te.mockReset()
+      tc.mockReset()
+    })
+
+    describe('still uses the message property of violations', () => {
+      it('if the locale is null and translation does not exist', () => {
+        expect(transformViolations(validationError, i18n)).toEqual({
+          'parent.property': ['some message', 'another message'],
+          property2: ['yet another message'],
+        })
+      })
+
+      it('if the translation for the locale is not in the response and translation does not exist', () => {
+        i18n.locale = 'it'
+        expect(transformViolations(validationError, i18n)).toEqual({
+          'parent.property': ['some message', 'another message'],
+          property2: ['fallback'],
+        })
+      })
+    })
+
+    describe('uses the frontend translation', () => {
+      it('if the key is available in the translations', () => {
+        te.mockReturnValue(true)
+        const translation = 'a translation'
+        tc.mockReturnValue(translation)
+
+        expect(transformViolations(validationError, i18n)).toEqual({
+          'parent.property': [translation, translation],
+          property2: [translation],
+        })
+      })
+
+      it('even if there are translations in the response', () => {
+        te.mockReturnValue(true)
+        const translation = 'a translation'
+        tc.mockReturnValue(translation)
+        i18n.locale = 'en'
+
+        expect(transformViolations(validationError, i18n)).toEqual({
+          'parent.property': [translation, translation],
+          property2: [translation],
+        })
+      })
+    })
+
+    describe('uses the translations in the response', () => {
+      it(' if the keys are not known in the frontend', () => {
+        te.mockReturnValue(false)
+        i18n.locale = 'fr-CH-scout'
+
+        expect(transformViolations(validationError, i18n)).toEqual({
+          'parent.property': ['some message', 'another message'],
+          property2: ['fr_CH_scout'],
+        })
+      })
+
+      it('uses the direct fallback', () => {
+        console.log('uses the direct fallback')
+        te.mockReturnValue(false)
+        i18n.locale = 'de-CH-scout'
+
+        expect(transformViolations(validationError, i18n)).toEqual({
+          'parent.property': ['some translation', 'some other translation'],
+          property2: ['fallback'],
+        })
+      })
+
+      it('uses fallbackLocale', () => {
+        console.log('uses fallbackLocale')
+        te.mockReturnValue(false)
+        i18n.locale = 'it'
+
+        expect(transformViolations(validationError, i18n)).toEqual({
+          'parent.property': ['some message', 'another message'],
+          property2: ['fallback'],
+        })
+      })
+    })
+  })
+})
+
+const unauthorizedError = {
+  name: 'ServerException',
+  response: {
+    status: 401,
+    data: {
+      detail: 'Unauthorized',
+    },
+  },
+}
+
+const validationError = {
```

Comment:
The selection of the wording makes it a bit hard to follow the tests. Should we be a bit more explicit in the message itself?

```
const validationError = {
  name: 'ServerException',
  response: {
    status: 422,
    data: {
      violations: [
        {
          message: 'some message (untranslated)',
          propertyPath: 'parent.property',
          i18n: {
            key: 'some.key',
            parameters: {},
            translations: {
              de: 'some message (de translated)',
            },
          },
        },
        {
          message: 'another message',
          propertyPath: 'parent.property',
          i18n: {
            key: 'some.other.key',
            parameters: {},
            translations: {
              de: 'another message (de translated)',
            },
          },
        },
        {
          message: 'yet another message',
          propertyPath: 'property2',
          i18n: {
            key: 'yet.other.key',
            parameters: {},
            translations: {
              fr: 'yet another message (fr translated)',
              fr_CH_scout: 'yet another message (fr_CH_scout translated)',
              en: 'yet another message (en translated)',
            },
          },
        },
      ],
    },
  },
}
```

Reply:
Went even a step further and made the tests independent of each other